controller/test/ControllerRTL_test.py

```diff
@@ -0,0 +1,127 @@
+'''
+=========================================================================
+ChannelRTL_test.py
+=========================================================================
+Simple test for Channel.
+
+Author : Cheng Tan
+  Date : Dec 2, 2024
+'''
+
+
+from pymtl3 import *
+from pymtl3.stdlib.test_utils import TestVectorSimulator
+from ..ControllerRTL import ControllerRTL
+from ...lib.basic.en_rdy.test_sinks import TestSinkRTL
+from ...lib.basic.en_rdy.test_srcs import TestSrcRTL
+from ...lib.basic.val_rdy.SourceRTL import SourceRTL as TestValRdySrcRTL
+from ...lib.basic.val_rdy.ifcs import SendIfcRTL as ValRdySendIfcRTL
+from ...lib.basic.val_rdy.ifcs import RecvIfcRTL as ValRdyRecvIfcRTL
+from ...lib.messages import *
+from ...noc.PyOCN.pymtl3_net.ocnlib.test.stream_sinks import NetSinkRTL as TestNetSinkRTL
+import pytest
+
+
+#-------------------------------------------------------------------------
+# TestHarness
+#-------------------------------------------------------------------------
+
+class TestHarness(Component):
+
+  def construct(s, MsgType, PktType, AddrType, src_msgs, sink_msgs, src_pkts, sink_pkts):
+
+    cmp_fn = lambda a, b : a.payload == b.payload
```

Comment:
What is `cmp_fn` used for?

Reply:
used to compare with the golden reference, i.e., RingFL.